### PR TITLE
Use the ".. deprecated::" directive for deprecations

### PR DESCRIPTION
**Pull request by OskarStark**
Refs #11154 

## Todo:

- [x] We need no ensure the correct html markup
- [x] @symfony/docs-team please review the new documentation about "[how/when to use this new directive](https://github.com/symfony/symfony-docs/commit/2b0504d9aeea9c523ccb5b85fb19962d25a6d198)"

### CODE REVIEW

**Review by wouterj — APPROVED** [review]

**Comment by wouterj** [comment]
Approved! And added "on hold", as we first have to make sure @javiereguiluz adds nice styling for this on symfony.com. The deprecated directive is rendered the same way as the versionadded (including the `.versionmodified` class), except that it's main class is `deprecated`. So it should be fairly easy to reuse the versionadded styles and only modify the color (i.e. some blackish)

File: contributing/documentation/format.rst [hunk 1]
```diff
@@ -168,35 +168,47 @@ of the linked resource (``namespace``, ``class`` or ``method``):
...
 
-If you're documenting a behavior change, it may be helpful to *briefly* describe
-how the behavior has changed:
+If you are documenting a behavior change, it may be helpful to *briefly*
+describe how the behavior has changed:
 
 .. code-block:: rst
 
-    .. versionadded:: 3.4
+    ..  versionadded:: 3.4
```
Comment (javiereguiluz) [inline contributing/documentation/format.rst]: This extra white space should be removed. Thanks.

**Comment by javiereguiluz** [comment]
There is a pending PR on symfony.com private repo to add the styles:

### Before

before

### After

after

And in dark mode:

after-dark-theme

**Comment by wouterj** [comment]
Nice!

**Comment by OskarStark** [comment]
Wow this looks amazing 🥰👍🏻

File: contributing/documentation/format.rst (hunk 1 above)
Comment (OskarStark) [reply contributing/documentation/format.rst]: Done

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
The style changes were merged on symfony.com, so we can now merge this. Thanks Oskar!

**Pull request closed without merging**